napi_error_handling.md

```diff
@@ -0,0 +1,115 @@
+# Error handling
+
+N-API uses both return values and exceptions for error handling.  The sections
+which follow explain the approach for each case.
+
+## Return values
+
+All of the N-API functions share the same error handling pattern.  The
+return type of all API functions is ```napi_status```.
+
+The return value will be ```napi_ok``` if the request was successful and
+no uncaught exception was thrown. If an error occurred AND an exception
+was thrown, the napi_status value for the error will be returned.  If an
+exception was thrown, and no error occurred, ```napi_pending_exception```
+will be returned.
+
+In cases were a return value other than ```napi_ok``` or
+```napi_pending_exception``` is returned, you must call
+```napi_is_exception_pending()``` to check if an exception is pending.
+See the section on exceptions for more details.
+
+The full set of possible napi_status values is defined
+in ```napi_api_types.h``.
+
+The 'napi_status' return value provides an vm-independent representation of
+the error which occured.  In some cases it is useful to be able to get
+more detailed information, including a string representing the error as well as
+vm(engine)-specific information.  **NOTE:**  you should not rely on the
+content or format of any of the extended information as it is
+not subject to SemVer and may change at any time. It is intended
+only for logging purposes.
+
+In order to retrieve this information, the following method is provided:
+
+```C
+NODE_EXTERN const napi_extended_error_info* napi_get_last_error_info();
+```
+and the format of the ```napi_extended_error_info``` structure is as follows:
+
+```C
+struct napi_extended_error_info {
+  const char* error_message;
```

Comment:
Currently the error message is VM-independent, because it is just a definition of the `napi_status` value. But I don't know if we want to guarantee that.